src/preprocessing/passes/ackermann.cpp

```diff
@@ -187,9 +188,110 @@ void collectFunctionsAndLemmas(FunctionToArgsMap& fun_to_args,
 
 /* -------------------------------------------------------------------------- */
 
+/* Given a minimum capacity for an uninterpreted sort, return the size of the
+ * new BV type */
+size_t getBVSkolemSize(size_t capacity)
+{
+  return static_cast<size_t>(log2(capacity)) + 1;
+}
+
+/* Given the lowest capacity requirements for each uninterpreted sort, assign
+ * a sufficient bit-vector size. Get the converting map */
+void collectUSortsToBV(const vector<TNode>& terms,
```

Comment:
There are actually variables, not arbitrary terms. This is crucial for the correctness of the minimum capacity, so I would rename this variable to `vars`, and make sure that these variables are not called terms throughout the conversion.